README.md

```diff
@@ -20,19 +20,107 @@ Functionalities:
+
+To set an evaluation context, you should use the `setEvaluationContextAndWait` method on the OpenFeature SDK.
+
+```kotlin
+val userId: String = "..." // get the user id from your app
+val evaluationContext = ImmutableContext(
+    targetingKey = userId,
+    attributes = mapOf(
+    "user_id" to Value.String(userId),
+    "country" to Value.String("SE"),
+    "product" to Value.String("productId123")
+))
+OpenFeatureAPI.setEvaluationContextAndWait(evaluationContext)
+```
+
+> [!NOTE]
+> When you set a new evaluation context, the Confidence provider will automatically pick it up, 
+> fetch new flag values and use them for all subsequent flag evaluations.
+
+
+### Resolving flags
+
+**Once the flags are fetched and activated**, you can access their value using the different methods on the OpenFeature `Client`.
+
+The `get<Type>Value` method returns a value of the specified type, where `<Type>` or the default value specified.
+
+The `get<Type>Details` method returns a typed `FlagEvaluationDetails` object that contains the `value` of the flag, the `reason` for the value returned, and the `variant` selected.
+In the case of an error, the default value will be returned and the `errorCode` and `errorMessage` contain information about the error.
+
+The method `get<Type>Value` will simply return the assigned value or the default.
+
+```kotlin
+val client = OpenFeatureAPI.getClient()
+
+val message: String = client.getStringValue("flag-name.message", "default message")
+val messageFlag: FlagEvaluationDetails<String> = client.getStringDetails("flag-name.message", "default message")
+
+val messageValue = messageFlag.value
+// message and messageValue are the same
+```
+
+> [!IMPORTANT]
+> You should note that all feature flags in Confidence are complex objects implemented by the `ConfidenceValue.Struct` class but fear not! 
+> You can use a dot-notation to access properties within your flag object for ease of use.
+> Above we show how to access a string property from a flag. In that example the actual flag object is called `flag-name` and we access the string property `message` using the dot notation: `flag-name.message`.
+
+### Initialization strategy
+`initialisationStrategy` is a way to decide how Confidence should act when the provider is being set.
+- `ActivateAndFetchAsync` will make the last fetched flags available immediately and then trigger a background fetch to update the flags for future sessions.
+- `FetchAndActivate` will fetch the flags from the Confidence backend, stores the result on disk, and make the same data ready for the Application to be consumed.
+  If freshness of flags are the most important aspect for your use case, this is the strategy to go for. However, this will introduce a delay in making flags available to the application.
```

Comment:
This could be confusing, we might want to spend a few extra paragraph to show (maybe in code?) how to wait for a Provider to be ready and when it's safe to start evaluating in the different strategy scenarios